Do not merge - blocked: TS Models (#2281)
mongodb/docs-realm: source/sdk/node/examples/define-a-realm-object-model.txt

   .. literalinclude:: /examples/generated/node/open-and-close-a-local-realm.snippet.open-local-realm-with-car-schema.js
     :language: javascript

.. _node-define-objects-using-a-class:

Define Realm Object Types Using a Class
---------------------------------------

.. tabs-realm-languages::

   .. tab::
      :tabid: typescript

>       You can declare a Realm Object Model using TypeScript property types

Reviewer: Are "Realm Object Model" and "Realm Object Schema" the same thing? We seem to use "schema" far more frequently in the Node and React Native SDK docs. If they're equivalent, I suggest rewriting all instances of "object model" to "object schema".

This page in particular has a Car schema with "model" as a property, so that could be a bit confusing - readers could conflate the two things.

Reviewer: Ha! I just noticed in other files you purposefully changed from "Schema" to "Model". Disregard this suggestion and the following ones about model -> schema if we're in fact moving the other direction.

         :language: javascript  
            
      Pass the class itself to the schema property of the :js-sdk:`Realm.Configuration
      <Realm.html#~Configuration>` object when opening a realm. You can then
      :doc:`read data </sdk/node/examples/read-and-write-data>` normally. Class based
      models call :js-sdk:`Realm.create() <Realm.html#create>` when you call their
      constructor, so you do not need to explicitly call ``Realm.create()``.
      
      .. literalinclude:: /examples/generated/node/define-a-realm-object-schema.snippet.define-a-realm-object-schema-js-classes-open-and-access-properties.js
         :language: javascript

      You can optionally add a constructor in your class. This may be useful if
      you need to set a value based on some calculation. In the following
      example, we define a constructor and round the ``miles`` travelled to the
      nearest integer.

      .. literalinclude:: /examples/generated/node/define-a-realm-object-schema.snippet.define-a-realm-object-schema-define-js-classes-with-constructor.js
         :language: javascript

Supported TypeScript Types
--------------------------

> This plugin supports standard TypeScript types wherever possible to make

Reviewer: "This plugin" is a bit vague. Unless readers are already familiar with it, they won't know that we support TypeScript Types only through the Realm Babel Plugin. It's mentioned in the note above, but that's easily missed. I recommend spelling it out here again.

```suggestion
The Realm Babel Plugin supports standard TypeScript types wherever possible to make
```

defining your model feel like a native TypeScript experience. Some Realm types
do not have a direct TypeScript equivalent. For example, ``double``, ``int``,

> and ``float`` are all represented by ``number`` in TypeScript). In these cases,

Reviewer: Typo: standalone closing parenthesis.
```suggestion
and ``float`` are all represented by ``number`` in TypeScript. In these cases,
```

use the types provided by ``Realm.Types``.

We recommend using TypeScript types wherever possible and using ``Realm.Types``
where you cannot.

> The top-level Realm namespace and ``Realm.Types`` exports various types, such as

Reviewer: Count and verb disagreement: Realm namespace and ``Realm.Types`` are two things, so the verb should agree with a plural noun form.
```suggestion
The top-level Realm namespace and ``Realm.Types`` export various types, such as
```
